### PR TITLE
Fix signaling cache and GStreamer sample build on Windows

### PR DESCRIPTION
**Pull request by niyatim23**
*Issue #, if available:*

*What was changed?*
Modified the Windows CI and deserialization logic in Signaling caching

*Why was it changed?*
Signaling cache was broken on Windows. This change would fix it and some tests that were disabled on Windows

*How was it changed?*
By modifying the readFile in the PIC layer and fixing the deserialize function on Windows by introducing an additional check

*What testing was done for the changes?*
The tests pass in the CI. Ran the sample application to make sure the cache is being used instead of the describe call

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/source/Signaling/FileCache.c [hunk 1]
```diff
@@ -38,7 +38,7 @@ STATUS deserializeSignalingCacheEntries(PCHAR cachedFileContent, UINT64 fileSize
     pCurrent = cachedFileContent;
     remainingSize = (UINT32) fileSize;
     /* detect end of file */
-    while (remainingSize > MAX_SIGNALING_CACHE_ENTRY_TIMESTAMP_STR_LEN) {
+    while (STRLEN(pCurrent) > 0 && remainingSize > MAX_SIGNALING_CACHE_ENTRY_TIMESTAMP_STR_LEN) {
```
Comment (hassanctech) [inline src/source/Signaling/FileCache.c:41]: Can we use `STRNLEN` instead of `STRLEN` since the latter is unsafe.

Comment (niyatim23) [reply src/source/Signaling/FileCache.c]: Done

File: src/source/Signaling/FileCache.c [hunk 2]
```diff
@@ -38,7 +38,7 @@ STATUS deserializeSignalingCacheEntries(PCHAR cachedFileContent, UINT64 fileSize
     pCurrent = cachedFileContent;
     remainingSize = (UINT32) fileSize;
     /* detect end of file */
-    while (remainingSize > MAX_SIGNALING_CACHE_ENTRY_TIMESTAMP_STR_LEN) {
+    while (STRNLEN(pCurrent, MAX_SERIALIZED_SIGNALING_CACHE_ENTRY_LEN * 32) > 0 && remainingSize > MAX_SIGNALING_CACHE_ENTRY_TIMESTAMP_STR_LEN) {
```
Comment (hassanctech) [inline src/source/Signaling/FileCache.c:41]: Instead of `32` use `MAX_SIGNALING_CACHE_ENTRY_COUNT` because it could change.

**Review by hassanctech — APPROVED** [review]
LGTM

**Pull request merged**